cmd/cmd_util.go

```
@@ -0,0 +1,152 @@
+	var err error
+	for _, pvc := range *pvcs {
+		pv, err := kubernetes.GetPV(client, pvc.Spec.VolumeName)
+		if err != nil {
+			return fmt.Errorf("failed to get PV object with name %s: %v", pvc.Spec.VolumeName, err)
+		}
+		logger.DefaultLog("PV found %v ", pv)
+		err = kubernetes.UpdateReclaimPolicy(client, pv)
+		if err != nil {
+			return fmt.Errorf("failed to update ReclaimPolicy for PV object %s: %v", pvc.Spec.VolumeName, err)
+		}
+		// Retrive old ceph volume name from PV
+		rbdImageName := kubernetes.GetFlexVolumeName(pv)
+		if rbdImageName == "" {
+			return fmt.Errorf("rbdImageName cannot be empty in Flex PV object: %v", pv)
+		}
+		logger.DefaultLog("flex rbdImageName name %v ", rbdImageName)
+
+		// Delete the PVC object
+		err = kubernetes.DeletePVC(client, &pvc) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+		if err != nil {
+			// TODO have an option to revert back if some Delete fails?
+			return fmt.Errorf("failed to Delete PVC object %s : %v", pvc.Name, err)
+		}
+		// Create new PVC with same name in destination storageclass
+		csiPVC := kubernetes.GenerateCSIPVC(destinationStorageClass, &pvc) // nolint:gosec // skip gosec as pvc is accessed via it's reference.
+		logger.DefaultLog("structure of the generated PVC in destination Storageclass \n %v ", csiPVC)
+
+		// Create new CSI PVC
+		csiPV, err := kubernetes.CreatePVC(client, csiPVC, 5)
+		if err != nil {
+			return fmt.Errorf("failed to Create CSI PVC object %s : %v", pvc.Name, err)
+		}
+		logger.DefaultLog("new PVC with same name created in CSI. Persistent Volume: %v ", csiPV)
+
+		// Extract new volume name from CSI PV
+		csiRBDImageName := kubernetes.GetCSIVolumeName(csiPV)
+		if csiRBDImageName == "" {
+			return fmt.Errorf("csiRBDImageName cannot be empty in PV object %v", csiPV)
+		}
+		logger.DefaultLog("csi new volume name: %v ", csiRBDImageName)
+		poolName := kubernetes.GetCSIPoolName(csiPV)
```

ditto why not return an error?